proto/Cargo.toml

```diff
@@ -26,6 +26,7 @@ tracing.workspace = true
 uuid.workspace = true
 serde = { workspace = true, optional = true }
 thiserror.workspace = true
+num_enum = "0.7.1"
```

Comment:
Question: typically I would avoid bringing a new proc macro crate / utility library unless it’s really worth it. In this case, bringing in all of this just for one enum doesn’t sounds very compelling. That being said, the heaviest dependency, syn, is already part of the dependency graph (via thiserror and serde), so it’s not too bad. I’m not asking you to remove it if you prefer using it and if @Firstyear is good with this additional dependency.

Reply:
My attitude is "bring in as few deps as possible, unless they really help and solve a problem that otherwise would be painful. 

I think in this case, there may be other places where num_enum could be used in the project, so it may be worth and as you say, it's already got most of it's deps in the code base. So I'm okay with it here.

Reply:
updated